### PR TITLE
Enable `merge` write disposition for `athena` Iceberg

### LINKED ISSUE #633: enable merge on ICEBERG via Athena
**Background**
See **rationale** below for why iceberg/open table. We can support it easily via existing destinations: Athena and Snowflake.
- snowflake ICEBERG https://www.snowflake.com/blog/iceberg-tables-powering-open-standards-with-snowflake-innovations/ - still in private preview so we skip that
- athena - supports ICEBERG and actually our destination is already using it for dlt tables

Why not spark:
- too much work for this ticket.

 **Requirements**
1. * [x] add a new table hint **table_format** that will tell destination which table format to use
2. * [x] for Athena: add global destination setting to set all tables to ICEBERG
3. * [ ] enable merge support for iceberg tables on athena

**Implementation Notes**
* Bulk loading into iceberg on athena can only be done from one table to another. So we need to retain our code for defining regular athena tables and additionally add iceberg tables to copy from the first tables. 
* Iceberg has nice "merge into ... on" sql directives that we can hopefully use for the merge write disposition
* Do we need to run maintenance commands that clean up the underlying parquet files on iceberg or should we leave this to the user?

**Rationale**
Just few points on why iceberg is a game changer on data lakes:
* it allows evolving the partitions
e.g. often data engineers design partitioning based on few use cases that they change over time, and Iceberg allows to just change those partition definitions, thanks to hidden partition (without rewriting the all dataset, that's the case for pure parquet)
* it has build in procedures to optimize the dataset
are you aware of the small file problem in data-lakes? if so iceberg has a build in function to compact "objects" and improve performance and cost. Just a note on that: I wrote quite some compaction procedures on my life, and I believe that Iceberg make that way of compacting data standardised (same as DeltaLake)
* using disposition like Merge > e.g. you can upsert by id
* being able to delete data from a dataset - that's what I mentioned already, and that's why I choose Iceberg to ingest sensitive data from Kakfa for example.
this decision came after leading a GDPR project that allow to remove sensitive data on a data lake - again game changer that make data engineers life much easier.

last but not least: Iceberg is becoming one of the facto open table format in the data landscape:
* Redshift now supports it (read only in preview - I net to test it)
* Snowflake supports it in preview (read/write)
* BiqQuery should support it (read only? - not sure)
* Trino supports it (read/write)
* The Athena supports it too - as it's an AWS fork of Trino (on engine v3)
* Spark supports it - the most complete engine that support iceberg
* Flink supports it - (read/write)
* Dremio Support it - read for sure, not sure about writes
* Clickhouse support it - only read
* DuckDB - support  it - only read
* Java client - read and write
* python client - read only for now
* rust client - coming soon
* go client - coming soon

### PR DESCRIPTION
**Pull request by jorritsandbrink**
### Description
This PR enables the `merge` write disposition (both `delete-insert` and `scd2` strategies) for the `athena` destination with `iceberg` table format.

for `delete-insert`:
- uses regular tables instead of temporary tables because Athena does not support those
    - these tables are created in the staging schema
    - each run, the previous' run table(s) get dropped and new table(s) for that run are created—hence, a version of these tables will always be visible in the staging schema (not super elegant, but not a big problem either I'd say)

### Related Issues

- Closes #633 

### Additional Context
Discussion on how we end up using this approach on this PR: https://github.com/dlt-hub/dlt/pull/1294

### CODE REVIEW

File: dlt/destinations/impl/athena/athena.py [hunk 1]
```diff
@@ -154,6 +160,64 @@ def __init__(self) -> None:
...
+            staging_root_table_name,
+            sql_client,
+            primary_keys,
+            unique_column,
+            dedup_sort,
+            condition,
+            condition_columns,
+        )
+        # DROP needs backtick as escape identifier
+        sql.insert(0, f"""DROP TABLE IF EXISTS {temp_table_name.replace('"', '`')};""")
```
Comment (rudolfix) [inline dlt/destinations/impl/athena/athena.py:198]: yes, and there's PR (#998 ) that allows to escape for DDL and DML separately

**Review by rudolfix — APPROVED** [review]
LGTM!

File: docs/website/docs/dlt-ecosystem/destinations/athena.md [hunk 2]
```diff
@@ -137,6 +137,13 @@ force_iceberg = "True"
 
 For every table created as an iceberg table, the Athena destination will create a regular Athena table in the staging dataset of both the filesystem and the Athena glue catalog, and then copy all data into the final iceberg table that lives with the non-iceberg tables in the same dataset on both the filesystem and the glue catalog. Switching from iceberg to regular table or vice versa is not supported.
 
+#### `merge` support
+The `merge` write disposition is supported for Athena when using iceberg tables.
+
+> Note that:
+> 1. there is a risk of tables ending up in inconsistent state in case a pipeline run fails mid flight, because Athena doesn't support transactions, and `dlt` uses multiple DELETE/UPDATE/INSERT statements to implement `merge`,
```
Comment (toddy86) [inline docs/website/docs/dlt-ecosystem/destinations/athena.md:144]: Why not use the `merge` syntax that is available in Athena?

Comment (jorritsandbrink) [reply docs/website/docs/dlt-ecosystem/destinations/athena.md:144]: Full discussion is here: https://github.com/dlt-hub/dlt/pull/1294.

_TLDR_: we're planning to add a new merge strategy based on the MERGE statement, but this isn't properly specced yet. We want consistent behavior accross different destinations, hence the implementation of `delete-insert` for `athena` now. MERGE-based `merge` for `athena` will come together with the other destinations.

**Pull request merged**